### ISSUE TITLE
🚀 Feature: create private repos

### ISSUE THREAD

**Issue by johnnyreilly** [JoshuaKGoldberg/create-typescript-app#1618]
### Bug Report Checklist

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I'm experimenting with creating a new project.  For now I want it to be private, I'll probably make it public later when I'm happy with what i've got.  At present, the only type of repos you can create are public.  Would it be good to be able to create private repos as well?

### Additional Info

One thing to bear in mind, is that some of the bundled tooling (Renovate etc) will only work on public repos for free I think.  That may be an issue - or maybe something that can enabled later?

**Comment by JoshuaKGoldberg** [comment]
Agreed, this is definitely something that should be supported. I don't know what the full list of changes would be, so that'd be good to figure out. Once that's here I'd be pretty confident this will be accepting PRs.
